fud2/fud-core/src/exec/driver.rs

```diff
@@ -455,29 +534,22 @@ impl DriverBuilder {
     }
 }
 
+#[derive(Debug, PartialEq, PartialOrd, Clone)]
+pub enum IO {
+    // we order these so when they are sorted, StdIO comes first
+    // the u32 is a rank used to know which files should be read from stdin first
```

Comment:
The idea behind this is confusing and I don't think good design.

`StdIO(i, path)` says "path was the `i`th input (or `i`th output) a `Request`." `i` is a `usize` because it represents an index. This information is useful for knowing which file should be read into first from `stdin` or output first to `stdout`. 

However, `IO` doesn't have any connection to a `Request` so it just becomes some confusing poorly specified order. 

Having `IO::StdIO` be a tuple is bad. Really what I want is an ordered list. This ordering information should be stored in the `Plan` by having an "input" array ordered by which inputs come first in the `Request` used to generate the `Plan`. Similarly the `Plan` will have a "results" array ordered by which outputs comes first in the `Request`.